src/Akka.Hosting/AkkaHostedService.cs

```diff
@@ -47,6 +47,19 @@ public virtual async Task StartAsync(CancellationToken cancellationToken)
                 async Task TerminationHook()
                 {
                     await ActorSystem.WhenTerminated.ConfigureAwait(false);
+                    
+                    /*
+                     * Set a non-zero exit code in the event that we get a known, confirmed unclean shutdown
+                     * from the ActorSystem / CoordinatedShutdown
+                     */
+                    switch (CoordinatedShutdown.ShutdownReason)
+                    {
+                        case CoordinatedShutdown.ClusterDowningReason _:
```

Comment:
We know for sure there are errors with these two shutdown reasons - all of the others are uncertain. So only return a non-zero exit code when we are 100% certain.

Reply:
We might want to differentiate these exit codes in the future.

Reply:
should we add `ClusterJoinUnsuccessfulReason`?

Reply:
> We might want to differentiate these exit codes in the future.

I think that's a bit of YAGNI - in this instance we just need to return something non-zero so the Windows Service supervision system knows the process didn't exit cleanly.

Reply:
The logs will make it clear what actually happened